python/cudf/cudf/core/dataframe.py

```diff
@@ -5126,26 +5126,35 @@ def _reduce(
         if level is not None:
             raise NotImplementedError("level parameter is not implemented yet")
 
-        if numeric_only not in (None, True):
-            raise NotImplementedError(
-                "numeric_only parameter is not implemented yet"
+        source = self
+        if numeric_only:
+            numeric_cols = (
+                name
+                for name in self._data.names
+                if is_numeric_dtype(self._data[name])
             )
-        axis = self._get_axis_from_axis_arg(axis)
+            source = self._get_columns_by_label(numeric_cols)
+            if source.empty:
+                return source.astype("float64")
+
+        axis = source._get_axis_from_axis_arg(axis)
 
         if axis == 0:
             try:
                 result = [
-                    getattr(self._data[col], op)(**kwargs)
-                    for col in self._data.names
+                    getattr(source._data[col], op)(**kwargs)
+                    for col in source._data.names
                 ]
             except AttributeError:
-                raise TypeError(f"cannot perform {op} with type {self.dtype}")
+                raise TypeError(
+                    f"cannot perform {op} with type {source.dtype}"
```

Comment:
This isn't a bug related to your PR, but I think it's worth fixing it here anyway: if someone attempts to perform an aggregation on a column type for which that aggregation is invalid, it will raise an `AttributeError`, but it looks like our error message in the `TypeError` that we reraise will fail because `DataFrame.dtype` doesn't exist. We should probably change this message to something like:
```suggestion
                    f"Not all column dtypes support op {op}"
```